tests/app/submitter/test_convert_payload_0_0_1.py

```diff
@@ -235,6 +237,45 @@ def test_answer_with_multiple_instances(self):
             # Check the converter correctly
             self.assertEqual(answer_object['data']['003'], ['0', '1', '2'])
 
+    def test_answer_without_qcode(self):
+        with self._app.test_request_context():
+            user_answer = [create_answer('GHI', 'String test + !', group_id='group-1', block_id='block-1')]
+
+            questionnaire = {
+                'survey_id': '021',
+                'data_version': '0.0.1',
+                'sections': [
+                    {
+                        'id': 'section-1',
+                        'groups': [
+                            {
+                                'id': 'group-1',
+                                'blocks': [
+                                    {
+                                        'id': 'block-1',
+                                        'questions': [
+                                            {
+                                                'id': 'question-2',
+                                                'answers': [
+                                                    {
+                                                        'id': 'GHI',
+                                                        'type': 'TextField'
+                                                    }
+                                                ]
+                                            }
+                                        ]
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                ]
+            }
+
+            routing_path = [Location(group_id='group-1', group_instance=0, block_id='block-1')]
+
+            convert_answers(self.metadata, QuestionnaireSchema(questionnaire), AnswerStore(user_answer), routing_path)
+
```

Comment:
Maybe worth asserting length of data to verify that it got ignored?. Current check will verify that we didn't run into a KeyError only?